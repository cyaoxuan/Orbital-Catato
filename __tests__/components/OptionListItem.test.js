import {
    OptionListItem,
    OptionListItemIcon,
    SwitchListItem,
} from "../../app/components/OptionListItem";
import { cleanup, fireEvent, render } from "@testing-library/react-native";

afterEach(cleanup);

describe("<OptionListItem />", () => {
    it("renders successfully", () => {
        const { getByText } = render(<OptionListItem title="Option" />);
        const listItem = getByText("Option");
        expect(listItem).toBeDefined();
    });

    it("renders correctly", () => {
        const tree = render(<OptionListItem />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("calls onPress callback when item is pressed", () => {
        const onPressMock = jest.fn();
        const { getByText } = render(
            <OptionListItem title="Option" onPress={onPressMock} />
        );
        const listItem = getByText("Option");
        fireEvent.press(listItem);
        expect(onPressMock).toHaveBeenCalled();
    });
});

describe("<OptionListItemIcon />", () => {
    it("renders successfully", () => {
        const { getByText } = render(
            <OptionListItemIcon title="Option" iconName="checkmark" />
        );
        const listItem = getByText("Option");
        expect(listItem).toBeDefined();
    });

    it("renders correctly", () => {
        const tree = render(<OptionListItemIcon />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("calls onPress callback when item is pressed", () => {
        const onPressMock = jest.fn();
        const { getByText } = render(
            <OptionListItemIcon
                title="Option"
                iconName="checkmark"
                onPress={onPressMock}
            />
        );
        const listItem = getByText("Option");
        fireEvent.press(listItem);
        expect(onPressMock).toHaveBeenCalled();
    });
});

describe("<SwitchListItem />", () => {
    it("renders successfully", () => {
        const { getByText } = render(
            <SwitchListItem title="Option" value={true} />
        );
        const listItem = getByText("Option");
        expect(listItem).toBeDefined();
    });

    it("renders correctly", () => {
        const tree = render(<SwitchListItem />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("calls onValueChange callback when switch value is changed", () => {
        const onValueChangeMock = jest.fn();
        const { getByRole } = render(
            <SwitchListItem
                title="Option"
                value={true}
                onValueChange={onValueChangeMock}
            />
        );
        const switchInput = getByRole("switch");
        fireEvent(switchInput, "valueChange", false);
        expect(onValueChangeMock).toHaveBeenCalledWith(false);
    });
});
