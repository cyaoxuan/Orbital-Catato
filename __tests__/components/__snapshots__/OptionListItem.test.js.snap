// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptionListItem /> renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      false,
      [
        {
          "paddingRight": 24,
          "paddingVertical": 8,
        },
        {
          "borderRadius": 16,
          "height": 65,
          "justifyContent": "center",
          "padding": 16,
        },
      ],
    ]
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "marginVertical": 6,
        "width": "100%",
      }
    }
  >
    <View
      style={
        [
          {
            "paddingLeft": 16,
          },
          {
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 16,
              },
              {
                "color": "rgba(28, 27, 31, 1)",
              },
              {
                "fontFamily": "Nunito-Medium",
                "fontSize": 20,
              },
            ],
          ]
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 20,
            },
            undefined,
            {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`<OptionListItemIcon /> renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      false,
      [
        {
          "paddingRight": 24,
          "paddingVertical": 8,
        },
        {
          "borderRadius": 16,
          "height": 65,
          "justifyContent": "center",
          "padding": 16,
        },
      ],
    ]
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "marginVertical": 6,
        "width": "100%",
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 20,
            },
            undefined,
            {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      />
    </View>
    <View
      style={
        [
          {
            "paddingLeft": 16,
          },
          {
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 16,
              },
              {
                "color": "rgba(28, 27, 31, 1)",
              },
              {
                "fontFamily": "Nunito-Medium",
                "fontSize": 20,
              },
            ],
          ]
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 20,
            },
            undefined,
            {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`<SwitchListItem /> renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      false,
      [
        {
          "paddingRight": 24,
          "paddingVertical": 8,
        },
        {
          "borderRadius": 16,
          "height": 65,
          "justifyContent": "center",
          "padding": 16,
        },
      ],
    ]
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "marginVertical": 6,
        "width": "100%",
      }
    }
  >
    <View
      style={
        [
          {
            "paddingLeft": 16,
          },
          {
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 16,
              },
              {
                "color": "rgba(28, 27, 31, 1)",
              },
              {
                "fontFamily": "Nunito-Medium",
                "fontSize": 20,
              },
            ],
          ]
        }
      />
    </View>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#05668D"
      style={
        [
          {
            "height": 31,
            "width": 51,
          },
          {
            "transform": [
              {
                "scaleX": 1.2,
              },
              {
                "scaleY": 1.2,
              },
            ],
          },
        ]
      }
      tintColor="#05668D"
      value={false}
    />
  </View>
</View>
`;
